vars/rpmDashboardsDistValidation.groovy

```diff
@@ -0,0 +1,127 @@
+def call(Map args = [:]) {
+
+    def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    def bundleManifestObj = args.bundleManifestObj
+    def distFile = args.rpmDistribution         //Distribution file location
+    def name = bundleManifestObj.build.getFilename()   //opensearch-dashboards
+    def version = bundleManifestObj.build.version        //1.3.0
+    def architecture = bundleManifestObj.build.architecture
+    def plugin_names = bundleManifestObj.getNames();
+
+    def latestOpenSearchURL = "https://ci.opensearch.org/ci/dbc/Playground/tianleh-test/tianle-opensearch-build-3-22/$version/latest/linux/$architecture/rpm/dist/opensearch/opensearch-$version-linux-${architecture}.rpm"
+    def latestOpensearchDist = "$WORKSPACE/opensearch-$version-linux-${architecture}.rpm"
+    // Download the latest OpenSearch distribution with same version to be compatible with Dashboards.
+    sh("curl -SLO $latestOpenSearchURL")
+
+    // This is a reference meta data which the distribution should be consistent with.
+    def refMap = [:]
+    refMap['Name'] = name
+    refMap['Version'] = version
+    refMap['Architecture'] = architecture
+    refMap['Group'] = "Application/Internet"
+    refMap['License'] = "Apache-2.0"
+    refMap['Relocations'] = "(not relocatable)"
+    refMap['URL'] = "https://opensearch.org/"
+    // The context of meta data should be for OSD
+    refMap['Summary'] = "Open source visualization dashboards for OpenSearch"
+    refMap['Description'] = "OpenSearch Dashboards is the visualization tool for data in OpenSearch\n" +
+            "For more information, see: https://opensearch.org/"
+
+    //Validation for the Meta Data of distribution
+    println("Meta data validations start:")
```

Comment:
I would extract metadata validation into a class and write tests.

Reply:
Makes sense to me. Will be working on this.